cpp/src/groupby/groupby.cu

```diff
@@ -79,6 +79,31 @@ std::pair<std::unique_ptr<table>, std::vector<aggregation_result>> groupby::disp
 groupby::~groupby() = default;
 
 namespace {
+
+/**
+ * @brief Factory to construct empty result columns.
+ *
+ * Adds special handling for COLLECT_LIST/COLLECT_SET, because:
+ * 1. `make_empty_column()` does not support construction of nested columns.
+ * 2. Empty lists need empty child columns, to persist type information.
+ */
+struct empty_column_constructor {
+  column_view values;
+
+  template <aggregation::Kind k>
+  std::unique_ptr<cudf::column> operator()() const
+  {
+    using namespace cudf;
+
+    if constexpr (k == aggregation::Kind::COLLECT_LIST || k == aggregation::Kind::COLLECT_SET) {
+      return make_lists_column(
+        0, make_empty_column(data_type{type_to_id<offset_type>()}), empty_like(values), 0, {});
+    }
+
+    return make_empty_column(cudf::detail::target_type(values.type(), k));
```

Comment:
```suggestion
    return empty_like(value);
```
Might make sense for this to be an `empty_like()` if `values` can themselves be nested types (or if they might be in the future)

Reply:
> Might make sense for this to be an `empty_like()`...

"I'm sorry, Dave. I'm afraid I can't do that."  | ( o ) |

One problem is that the returned column has to be appropriate for the aggregation. E.g. `COUNT()` must return an `INT32` column, `MIN()` returns the input type `T`, `MEAN()` returns `FLOAT64` (except when it doesn't), etc.

I have a table that describes this in the sister PR to this one: https://github.com/rapidsai/cudf/pull/8274

But you do raise something that I hadn't considered. I'll try write a test that exercises what happens for aggregations on `LIST` columns.

Reply:
I've changed it so that for aggregations where that return the same type as the input column return `empty_like(values)`. Note that I still need the `make_empty_column(target_type(values.type(), k))` for other aggregations.